backend/dataline/services/connection.py

```diff
@@ -270,3 +270,55 @@ async def create_sas7bdat_connection(self, session: AsyncSession, file: UploadFi
         finally:
             # Clean up the temporary file
             os.unlink(temp_file_path)
+
+    async def refresh_connection_schema(self, session: AsyncSession, connection_id: UUID) -> ConnectionOut:
+        """
+        Refresh the schema of a connection. Flow of the function:
+        1. Get the connection details from the database
+        2. Get the latest schema information from the database (using DatalineSQLDatabase)
+        3. Create new ConnectionOptions with schema information from step 2
+        4. Fetch stored ConnectionOptions from the database and, if it exists, merge with new schema information
+        5. Sort schemas and tables by name
+        6. Update the connection with new options
+        """
+        connection = await self.connection_repo.get_by_uuid(session, connection_id)
+
+        # Get the latest schema information
+        db = await self.get_connection_details(connection.dsn)
+
+        # Create new ConnectionOptions with updated schema information
+        new_schemas = [
+            ConnectionSchema(
+                name=schema,
+                tables=[ConnecitonSchemaTable(name=table, enabled=False) for table in tables],
```

Comment:
why don't you set enabled here at once based on the existing schemas?

would be easier to follow the flow I think, and potentially less code

Reply:
Fetching the old tables' enabled statuses in this list comprehension becomes too complicated because you have to check if the schema exists and then you have to check if the table exists and they're all lists not dicts so you have to find the table in a list, from a potentially nonexistent schema in a list of schemas...

I tried doing it in another flow, added ~20 lines of code and the readability isn't much better. See this commit https://github.com/RamiAwar/dataline/commit/e3a6360711375c1196ece6ff37dba7d9a9182b4e (it's in a separate branch)

Reply:
Done 😃